core/modules/filters/equals.js

```diff
@@ -0,0 +1,63 @@
+/*\
+title: $:/core/modules/filters/equals.js
+type: application/javascript
+module-type: filteroperator
+
+Filter operator for comparing text for equality
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+/*
+Export our filter function
+*/
+exports.equals = function(source,operator,options) {
+	var results = [],
+		fieldname = (operator.suffix || operator.operator || "title").toLowerCase();
+	if(operator.prefix === "!") {
+		if(operator.regexp) {
+			source(function(tiddler,title) {
+
+				var text = title ;
+					if(text !== null && !operator.regexp.exec(text)) {
+						results.push(title);
+					}
+			});
+		} else {
+			source(function(tiddler,title) {
+
+				var text = title ;
+					if(text !== null && text !== operator.operand) {
+						results.push(title);
+					}
+			});
+		}
+	} else {
+		if(operator.regexp) {
+			source(function(tiddler,title) {
+				
+				var text = title ;
+				if(text !== null && !!operator.regexp.exec(text)) {
```

Comment:
You can use `operator.regexp.text(text)` which returns a boolean to avoid having to cast an array to a boolean with `!!`

Reply:
Not knowing why Jeremy or the original author chose to go this route, I'll stick with the existing code.  In any event, I suspect that this part of the code will be seldom used since regular expressions as operators have been deprecated.

Reply:
To keep things simple, I'd be inclined not to bother with the support for regexp operands here (unless there's reason to believe that it's particularly useful in this context)